build_tools/build_ext.py

```diff
@@ -106,7 +106,9 @@ def run(self) -> None:
                 if isinstance(ext, CMakeExtension):
                     print(f"Building CMake extension {ext.name}")
                     # Set up incremental builds for CMake extensions
-                    setup_dir = Path(__file__).resolve().parent.parent
+                    setup_dir = Path(
+                        os.getenv("CMAKE_SETUP_DIR", Path(__file__).resolve().parent.parent)
+                    )
                     build_dir = setup_dir / "build" / "cmake"
```

Comment:
Some minor tweaks:
- Prefixing envvar with `NVTE_` helps avoid conflicts with CMake envvars
- We should build directly in the build dir if provided, not in a `build/cmake` subdirectory
```suggestion
                    build_dir = os.getenv("NVTE_CMAKE_BUILD_DIR")
                    if build_dir:
                        build_dir = Path(build_dir).resolve()
                    else:
                        root_dir = Path(__file__).resolve().parent.parent
                        build_dir = root_dir / "build" / "cmake"
```

Reply:
good suggestions, thanks